### PR TITLE
Initial test of Arduino Simulator.

### PR DESCRIPTION
**Pull request by jamessynge**
Not yet a test of sensors.py, but leading up to it.

### CODE REVIEW

File: peas/tests/serial_handlers/protocol_arduinosimulator.py [hunk 1]
```diff
@@ -417,5 +397,58 @@ def _update_break_state(self):
...
+        an stored in instance
+        """
+        expected = 'expected a string in the form "arduinosimulator://[?board=<name>]"'
+        parts = urllib.parse.urlparse(url)
+        if parts.scheme != "arduinosimulator":
+            raise Exception(
+                expected + ': got scheme {!r}'.format(parts.scheme))
+        params = {}
+        for option, values in urllib.parse.parse_qs(parts.query, True).items():
+            if option == 'board' and len(values) == 1:
```
[the remark this answers is not in the document]
Comment (jamessynge) [reply peas/tests/serial_handlers/protocol_arduinosimulator.py]: In order to test the readline timeout in SerialData.read, I might want to change rate at which the bytes are emitted. We may want to deliberately add an error sometimes to test the response. And you requested a possible change to values other than millis and report_num.

File: peas/tests/test_sensors.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    yield True
+    # Remove our test handlers.
+    serial.protocol_handler_packages.remove('serial_handlers')
+
+
+# --------------------------------------------------------------------------------------------------
+# Basic tests of FakeArduinoSerialHandler.
+
+def test_create(serial_handlers):
+    ser = rs232.SerialData(port='arduinosimulator://')
```
[the remark this answers is not in the document]
Comment (jamessynge) [reply peas/tests/test_sensors.py]: There is a "get('board', 'telemetry')" in _create_simulator, so it defaults to the telemetry board. OK?

**Comment by jamessynge** [comment]
p.s. we need to add checking of the code style in peas/

**Comment by wtgee** [comment]
> p.s. we need to add checking of the code style in peas/

Yes, good point. I'll do it right now.

**Pull request merged**